### ISSUE TITLE
warnings when building C code in build/c with make

### ISSUE THREAD

**Issue by nikohansen** [numbbo/coco#351]
On the Mac
```
[15:37:12] 1 c$ make
cc -c -g -ggdb -std=c89 -pedantic -Wall -Wextra -Wstrict-prototypes -Wshadow -Wno-sign-compare -Wconversion -o example_experiment.o example_experiment.c
cc -c -g -ggdb -std=c89 -pedantic -Wall -Wextra -Wstrict-prototypes -Wshadow -Wno-sign-compare -Wconversion -o coco.o coco.c
code-experiments/src/logger_biobj_avl_tree.c:119:35: warning: missing field 'next' initializer [-Wmissing-field-initializers]
const avl_node_t avl_node_0 = { 0 };
                                  ^
code-experiments/src/logger_biobj_avl_tree.c:120:35: warning: missing field 'head' initializer [-Wmissing-field-initializers]
const avl_tree_t avl_tree_0 = { 0 };
                                  ^
code-experiments/src/logger_biobj_avl_tree.c:121:45: warning: missing field 'deallocate' initializer [-Wmissing-field-initializers]
const avl_allocator_t avl_allocator_0 = { 0 };
                                            ^
3 warnings generated.
cc -g -ggdb -std=c89 -pedantic -Wall -Wextra -Wstrict-prototypes -Wshadow -Wno-sign-compare -Wconversion -o example_experiment coco.o example_experiment.o -lm  
```

**Comment by ttusar** [comment]
All these come from the AVL tree code that is not ours. What should I do with them?

**Comment by nikohansen** [comment]
Solved, see 12324cdca66e40ac8a75f52e8a1421d576089571
